internal/drivers/eventhub/eventhub.go

```diff
@@ -91,16 +106,36 @@ func (p *eventHubDriver) Process(event internal.DBChangeEvent) (bool, error) {
 	return false, nil
 }
 
+func (p *eventHubDriver) getKeys(record *util.Record, companyId string, locationId string) (string, string) {
+	key := fmt.Sprintf("dbchange.%s.%s.%s.%s.%s", record.Table, record.Operation, strWithDef(&companyId, "NONE"), strWithDef(&locationId, "NONE"), record.Id)
+	partitionKey := fmt.Sprintf("%s.%s.%s.%s", record.Table, strWithDef(&companyId, "NONE"), strWithDef(&locationId, "NONE"), record.Id)
```

Comment:
i think the only issue with this is that it presumes that you wont do any realtime data lookups to other related tables that could be behind. 

## example
suppose a customer A had a parent (ie fleet) B. if you updated A 5 times and then deleted B. These messages will get processed in parallel which means you might get into a state where B is deleted from the database before you finish processing messages from A which means you'll get an A record with a parent id B that doesnt exist.

 probably not that big of a deal i would guess based on the needs of EDS customers but worth pointing out 🤔

Reply:
your case won't happen with this change since all the records with the same primary key are processed in order since they will all has to the same partition. whereas before this change, the condition you describe could happen. right?

Reply:
my case actually is caused by partitioning by primary key. Updates to A will be sequential, as well as updates to B. But A and B wont be coordinated, you could (in theory) process all of B before A.